src/generator/AutoRest.Ruby.Azure/Model/CompositeTypeRba.cs

```diff
@@ -42,15 +50,49 @@ public override string GetBaseTypeName()
                 if (this.BaseModelType.Extensions.ContainsKey(AzureExtensions.ExternalExtension) ||
                     this.BaseModelType.Extensions.ContainsKey(AzureExtensions.AzureResourceExtension))
                 {
-                    typeName = "MsRestAzure::" + typeName;
+                    if(!nameRegEx.IsMatch(typeName) || !IsResourceModelsMatchStandardDefinition(this))
```

Comment:
minor: a space after if and before ( to keep the formatting consistent across codebase